### PR TITLE
[Shader Graph] Add Blackbody node 

### PR DESCRIPTION
**Pull request by alindmanUnity**
### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing?
This adds the blackbody node to the node library. 

---
### Release Notes
Please add any useful notes about the feature/fix that might be helpful for others.
Added node to graphics test scene for Input > Gradient. 

---
### Testing status
**Katana Tests**: First off we need to make sure the Katana SRP tests are green?

**Manual Tests**: What did you do?
- [x] Opened test project + Run graphic tests locally
- [x] Built a player
- [x] Checked new UI names with UX convention
- [x] Tested UI multi-edition + Undo/Redo
- [x] C# and shader warnings (supress shader cache to see them)
- [x] Checked new resources path for the reloader (in devloper mode, you have a button at end of resources that check the pathes)
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)
Added blackbody to Input scene tests 
InputNodes_LI

Alternative, launch Yamato (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/sg%252Fblackbody-node

Any test projects to go with this to help reviewers?

---
### Overall Product Risks
**Technical Risk**: None, Low, Medium, High? Low - adds new node to library to cover

**Halo Effect**: None, Low, Medium, High? None - does not change any existing code

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

**Comment by alindmanUnity** [comment]
Shader Graph automated tests are green

**Comment by alindmanUnity** [comment]
is going to conflict with #4435 we need to pick one to merge first and resolve the other after

File: com.unity.shadergraph/Documentation~/Blackbody-Node.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Blackbody Node
+
+## Description
```
Comment (sebastienlagarde) [inline com.unity.shadergraph/Documentation~/Blackbody-Node.md]: Hey, sorry to bother you with this but regarding color temperature you need to provide full details in the documentation about what are the various convention used for the node.

- What is the output color space (should be linear RGB)
- What is the white point used for the conversion (in game it must be D65)
- What is the CIE color space used (in game we prefer to used CIE 1931 2degree)

Below this is the fit color temperature that I have done in Unity (cpp side for hdrp) with max error with above convention. For shader I recommend something simpler, but we should used the same convention to match the output. i.e the output of this node in HDRP used as emissive value (i.e it use Luminance) on an unlit material must match a rect area light with the same temperature (which display luminance too).

// Given a correlated color temperature (in Kelvin), estimate the RGB equivalent. Curve fit error is max 0.008.
ColorRGBAf CorrelatedColorTemperatureToRGB(float temperature)
{
    float r, g, b;

    // Temperature must fall between 1000 and 40000 degrees
    // The fitting require to divide kelvin by 1000 (allow more precision)
    float kelvin = clamp(temperature, 1000.f, 40000.f) / 1000.0f;
    float kelvin2 = kelvin * kelvin;

    // Using 6570 as a pivot is an approximation, pivot point for red is around 6580 and for blue and green around 6560.
    // Calculate each color in turn (Note, clamp is not really necessary as all value belongs to [0..1] but can help for extremum).
    // Red
    r = kelvin < 6.570f ? 1.0f : clamp((1.35651f + 0.216422f * kelvin + 0.000633715f * kelvin2) / (-3.24223f + 0.918711f * kelvin), 0.0f, 1.0f);
    // Green
    g = kelvin < 6.570f ?
        clamp((-399.809f + 414.271f * kelvin + 111.543f * kelvin2) / (2779.24f + 164.143f * kelvin + 84.7356f * kelvin2), 0.0f, 1.0f) :
        clamp((1370.38f + 734.616f * kelvin + 0.689955f * kelvin2) / (-4625.69f + 1699.87f * kelvin), 0.0f, 1.0f);
    //Blue
    b = kelvin > 6.570f ? 1.0f : clamp((348.963f - 523.53f * kelvin + 183.62f * kelvin2) / (2848.82f - 214.52f * kelvin + 78.8614f * kelvin2), 0.0f, 1.0f);

    return ColorRGBAf(r, g, b, 1.f);
}

So just be sure to check convention of the snippet you have take. We can see this together next week.

Comment (alindmanUnity) [reply com.unity.shadergraph/Documentation~/Blackbody-Node.md]: Sorry for the late update, here's what I remember of the function:

- Output should be linear RGB.
- White point is in fact D65.
- I'm pretty sure this is CIE 1964 deg10.

Are those the proper formats for compatibility in HD? We can just make notes in the documentation if so.

Comment (sebastienlagarde) [reply com.unity.shadergraph/Documentation~/Blackbody-Node.md]: Hey, yes, it is the proper format

Comment (sebastienlagarde) [reply com.unity.shadergraph/Documentation~/Blackbody-Node.md]: good to add in the doc

**Pull request merged**